spiped/dispatch.c

```diff
@@ -301,6 +304,8 @@ dispatch_shutdown(void * dispatch_cookie)
 		events_timer_cancel(A->dnstimer_cookie);
 	if (A->T != NULL)
 		dnsthread_kill(A->T);
+	if (A->sa_b != NULL)
+		sock_addr_free(A->sa_b);
```

Comment:
Don't want to free this here; we free it in `main`.

Reply:
(Or rather, we *should* free it in `main`.  Looks like we're not doing that yet though.)

Reply:
Fixed.